samples/boards/nordic/system_off/sample.yaml

```diff
@@ -4,34 +4,100 @@ common:
+      - nrf5340dk/nrf5340/cpuapp
+      - nrf54l15dk/nrf54l15/cpuapp
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "system off demo"
+        - "Retained data not supported"
+        - "Entering system off; press sw0 to restart"
   sample.boards.nrf.system_off.nrf_retained:
-    build_only: true
     integration_platforms:
       - nrf52840dk/nrf52840
     platform_allow:
       - nrf52840dk/nrf52840
       - nrf52dk/nrf52832
     extra_configs:
       - CONFIG_APP_USE_NRF_RETENTION=y
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "system off demo"
+        - "Retained data: INVALID"
+        - "Boot count: 1"
+        - "Off count: 0"
+        - "Active Ticks:"
+        - "Entering system off; press sw0 to restart"
   sample.boards.nrf.system_off.retained_mem:
     extra_args: DTC_OVERLAY_FILE="boards/nrf54l15dk_nrf54l15_cpuapp_retained_mem.overlay"
-    build_only: true
     platform_allow:
       - nrf54l15dk/nrf54l15/cpuapp
     extra_configs:
       - CONFIG_APP_USE_RETAINED_MEM=y
       - CONFIG_RETAINED_MEM=y
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "system off demo"
+        - "Retained data: INVALID"
+        - "Boot count: 1"
+        - "Off count: 0"
+        - "Active Ticks:"
+        - "Entering system off; press sw0 to restart"
   sample.boards.nrf.system_off.grtc_wakeup:
-    build_only: true
     platform_allow:
       - nrf54l15dk/nrf54l15/cpuapp
     extra_configs:
       - CONFIG_GRTC_WAKEUP_ENABLE=y
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "system off demo"
+        - "Retained data not supported"
+        - "Entering system off; wait 2 seconds to restart"
+        - "system off demo"
+        - "Retained data not supported"
+        - "Entering system off; wait 2 seconds to restart"
+        - "system off demo"
+        - "Retained data not supported"
+        - "Entering system off; wait 2 seconds to restart"
+  sample.boards.nrf.system_off.retained_mem.grtc_wakeup:
+    extra_args: DTC_OVERLAY_FILE="boards/nrf54l15dk_nrf54l15_cpuapp_retained_mem.overlay"
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
+    extra_configs:
+      - CONFIG_APP_USE_RETAINED_MEM=y
+      - CONFIG_GRTC_WAKEUP_ENABLE=y
+      - CONFIG_RETAINED_MEM=y
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "system off demo"
+        - "Retained data: INVALID"
+        - "Boot count: 1"
+        - "Off count: 0"
+        - "Active Ticks:"
+        - "Entering system off; wait 2 seconds to restart"
+        - "system off demo"
+        - "Retained data: valid"
+        - "Boot count: 2"
+        - "Off count: 1"
+        - "Active Ticks:"
+        - "Entering system off; wait 2 seconds to restart"
```

Comment:
looks like all of this is about using the sample as a test -- testing that the sample *builds* correctly in its default configuration + optional configurations is of course fine, but it shouldn't be used as a test (see https://docs.zephyrproject.org/latest/samples/sample_definition_and_criteria.html#twister-should-be-able-to-build-every-sample)

Reply:
It is just verifying the output and it states in sample description that:
> Sample output can be validated by leveraging the harness_config regex option, wherever applicable.

Reply:
Exactly as @nordic-krch stated, it is just validation of sample - booting and basic logs from running.